src/repr/src/adt/numeric.rs

```diff
@@ -616,3 +616,22 @@ pub fn extract_typ_mod(typ_mod: &[u64]) -> Result<Option<u8>, anyhow::Error> {
         _ => unreachable!(),
     })
 }
+
+/// Used to do potentially lossy value-to-value conversions while consuming the input value. Useful
+/// for interoperability between Numeric and f64.
+pub trait LossyFrom<T>: Sized {
```

Comment:
I wasn't super happy with this approach. The issue is that you can't do `f64::from(i)` if `i` is type `i64` since the conversion is potentially lossy. Which means I can't use the `From<i64` trait on generic methods. So if anyone has a better approach please let me know. Also if there's a better place to put this please let me know.

Reply:
Also if you have a better name, since it's technically only lossy when used with `f64` and not `Numeric`, let me know.

Reply:
I think d155340 cleans this up slightly with the `DecimalLike` trait.